custom_components/pyscript/function.py

```diff
@@ -190,13 +190,18 @@ def service_has_service(cls, domain, name):
     async def service_call(cls, domain, name, **kwargs):
         """Implement service.call()."""
         curr_task = asyncio.current_task()
-        if "context" in kwargs and isinstance(kwargs["context"], Context):
-            context = kwargs["context"]
-            del kwargs["context"]
-        else:
-            context = cls.task2context.get(curr_task, None)
-
-        await cls.hass.services.async_call(domain, name, kwargs, context=context)
+        hass_args = {}
+        for keyword, typ, default in [
+            ("context", Context, cls.task2context.get(curr_task, None)),
+            ("blocking", bool, None),
+            ("limit", float, None),
+        ]:
+            if keyword in kwargs and isinstance(kwargs[keyword], typ):
+                hass_args[keyword] = kwargs.pop(keyword)
+            elif default:
+                hass_args[keyword] = default
+
```

Comment:
This is a nice compact way to process the `kwargs`.  I was thinking we should log an error if the keyword argument doesn't match the type, but since these potentially override service parameters with same name, we should probably just silently pass them along as you are doing.  So this look as is.